source/opt/debug_info_manager.cpp

```diff
@@ -446,6 +446,42 @@ bool DebugInfoManager::IsDeclareVisibleToInstr(Instruction* dbg_declare,
   return IsAncestorOfScope(instr_scope_id, decl_scope_id);
 }
 
+void DebugInfoManager::AddDebugValueWithIndex(uint32_t dbg_local_var_id,
+                                              uint32_t value_id,
+                                              uint32_t expr_id,
+                                              uint32_t index_id,
+                                              Instruction* insert_before) {
+  std::unique_ptr<Instruction> new_dbg_value(new Instruction(
```

Comment:
Why doesn't this re-use `AddDebugValue`?  Shouldn't it need to check the same things that `AddDebugValue` checks?  And now in #3511, `AddDebugValue` returns a boolean flag indicating success.  It seems odd that this would not be also expected in this function.

Reply:
`AddDebugValueWithIndex()` simply generates and inserts a single `DebugValue` while `AddDebugValue()` does the same thing for the multiple `DebugValue`. I updated `AddDebugValue()` to use `AddDebugValueWithIndex()`.

In terms of the boolean return of `AddDebugValue()`, it does not matter that much of this PR. It will follow #3511 when I merge it.

Reply:
OK, so now I have a question about `AddDebugValue()`.  It seems to be a low-level primitive, and yet it may (under some conditions) not actually add a `DebugValue()`.  This is confusing, as it would mean that a low-level helper decides whether or not to do something (despite the name of the function).  Perhaps rename the function to reflect this?  Something like `MaybeAdd...` or (better) make the caller responsible for calling `AddDebugValue` when it really means to add a value.

Reply:
I agree that these names are confusing. I just update `AddDebugValue()` to `AddDebugValueIfVarDeclIsVisible()`.